subsys/bluetooth/host/smp.c

```diff
@@ -3251,7 +3272,7 @@ static u8_t smp_pairing_random(struct bt_smp *smp, struct net_buf *buf)
 
 	atomic_set_bit(&smp->allowed_cmds, BT_SMP_DHKEY_CHECK);
 	atomic_set_bit(smp->flags, SMP_FLAG_DHCHECK_WAIT);
-	smp_send_pairing_random(smp);
+	return smp_send_pairing_random(smp);
 #endif /* CONFIG_BT_PERIPHERAL */
 
 	return 0;
```

Comment:
Dead return `0` here if CONFIG_BT_PERIPHERAL is defined, Likely build warning/failure here.
It would be a failure if you end up on this return with no peripheral defined. 
```
#else
return BT_SMP_ERR_UNPSECIFIED
#endif /*CONFIG_BT_PERIPHERAL */
```

Reply:
I guess we should be using BT_SMP_ERR_PAIRING_NOTSUPP since we cannot pair as peripheral.

Reply:
Sounds good.